src/main/java/org/example/bookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,34 @@
+package org.example.bookstore.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.dto.BookSearchParameters;
+import org.example.bookstore.model.Book;
+import org.example.bookstore.repository.SpecificationBuilder;
+import org.example.bookstore.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private static final String TITLE_KEY = "title";
+    private static final String AUTHOR_KEY = "author";
```

Comment:
better make these constants public and have only one constant and one place for future changing